File: core/src/main/java/io/nirahtech/erp/core/EmailAddress.java
package io.nirahtech.erp.core;

public final class EmailAddress {
    private final String username;
    private final String domain;

    public EmailAddress(String username, String domain) {
        this.username = username;
        this.domain = domain;
    }

    public String getUsername() {
        return username;
    }

    public String getDomain() {
        return domain;
    }

}

File: webapp/src/main/react/src/data/EmailAddress.ts
class EmailAddress {
    private readonly username: string;
    private readonly domain: string;

    constructor(username: string, domain: string) {
        this.username = username;
        this.domain = domain;
    }

    getUsername(): string {
        return this.username;
    }

    getDomain(): string {
        return this.domain;
    }

    toString(): string {
        return `${this.username}@${this.domain}`;
    }
}
export default EmailAddress;